server/scripts/dbNode.ts

```diff
@@ -0,0 +1,359 @@
+import { factory, formatFilename } from '../../shared/logging';
+import Identity from '../eventHandlers/pgIdentity';
+import { Pool } from 'pg';
+import _ from 'underscore';
+import format from 'pg-format';
+import {
+  createListener,
+  chainSupportedBy,
+  SubstrateTypes,
+  SubstrateEvents
+} from '@commonwealth/chain-events';
+
+import {
+  RabbitMqHandler,
+} from '@commonwealth/ce-rabbitmq-plugin';
+import { DATABASE_URI, HANDLE_IDENTITY } from '../config'
+import RabbitMQConfig from '../util/rabbitmq/RabbitMQConfig';
+
+const log = factory.getLogger(formatFilename(__filename));
+
+// TODO: RollBar error reporting
+
+// the number of the current worker
+const WORKER_NUMBER: number = Number(process.env.WORKER_NUMBER) || 0;
+
+// the total number of workers
+const NUM_WORKERS: number = Number(process.env.NUM_WORKERS) || 1;
+
+// The number of minutes to wait between each run -- rounded to the nearest whole number
+const REPEAT_TIME = Math.round(Number(process.env.REPEAT_TIME)) || 1;
+
+// counts the number of errors occurring for each chain - resets every 24 hours (to remove any temporary connection errors)
+// this is only meant to stop a chain from repeatedly causing errors every REPEAT_TIME
+let chainErrors: { [chain: string]: number } = {}
+
+// counts the number of mainProcess runs
+let runCount = 0
+
+// any fatal error is handle through here
+async function handleFatalError(error: Error, pool, chain?: string, type?: string): Promise<void> {
+  log.error(`${chain ? '[' + chain + ']: ' : ''}${String(error)}`);
+
+  if (chain && chain != 'erc20' && chainErrors[chain] >= 4) {
+    listeners[chain].unsubscribe();
+    delete listeners[chain];
+
+    const query = format(`UPDATE "Chains" SET "has_chain_events_listener"='false' WHERE "id"=%L`, chain)
+    try {
+      pool.query(query);
+    } catch (error) {
+      log.fatal(`Unable to disabled ${chain}`)
+    }
+  } else if (chain) ++chainErrors[chain];
+}
+
+// stores all the listeners a dbNode has active
+const listeners: { [key: string]: any} = {};
+
+// the function that executes every REPEAT_TIME
+async function mainProcess(producer: RabbitMqHandler, pool) {
+  // reset the chainError counts at the end of every day
+  if (runCount > 1440 / REPEAT_TIME) {
+    runCount = 1;
+    chainErrors = {}
+  } else {
+    ++runCount;
+  }
+
+  log.info('Starting scheduled process');
+
+  let query = `SELECT "Chains"."id", "substrate_spec", "url", "address", "base", "type", "network" FROM "Chains" JOIN "ChainNodes" ON "Chains"."id"="ChainNodes"."chain" WHERE "Chains"."has_chain_events_listener"='true';`;
```

Comment:
`pgIdentity.ts` is an exact replica (or should be) of `identity.ts` since the early plan for this new system (back when this script was still in the chain-events repo) was to let the nodes handle the identity events. As for using sequelize in dbNode, we can though I do believe it will utilize way more memory since all the models have to be loaded into the dyno's memory.  Leaving as is also makes it WAYYY easier to move away from heroku into a separate system somewhere if that's ever needed. So its up to you - we can convert to sequelize if needed and should be pretty simple.